src/main/java/net/dv8tion/jda/api/events/guild/scheduledevent/GuildScheduledEventUserAddEvent.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.dv8tion.jda.api.events.guild.scheduledevent;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.GuildScheduledEvent;
+import net.dv8tion.jda.api.entities.User;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Indicates that a {@link net.dv8tion.jda.api.entities.User User} has shown that they are intertested/have subscribed
+ * to a {@link net.dv8tion.jda.api.entities.GuildScheduledEvent GuildScheduledEvent}.
+ *
+ * Can be used to detect when someone has indicated that they are interested in an event and also retrieve their
+ * {@link net.dv8tion.jda.api.entities.User User} object as well as the {@link GuildScheduledEvent}.
+ */
+public class GuildScheduledEventUserAddEvent extends GenericGuildScheduledEventGatewayEvent
+{
+    private final long userId;
+
+    public GuildScheduledEventUserAddEvent(@Nonnull JDA api, long responseNumber, @Nonnull GuildScheduledEvent guildScheduledEvent, @Nonnull long userId)
+    {
+        super(api, responseNumber, guildScheduledEvent);
+        this.userId = userId;
+    }
+
```

Comment:
discord only sends the user id on this event, so I didn't implement a getUser() method as it would ne nullable and I find that these convenience methods can often confuse users that are not familiar with how caching works now. Personally I think its better to let the user get the user object himself to not be tricked into thinking the user is from the event and not the cache. 

Interested to hear what minn and/or dv8 think about this before going one way or the other.

Reply:
then @MinnDevelopment /  @DV8FromTheWorld them